test/unit/media-controller.spec.js

```diff
@@ -0,0 +1,160 @@
+import { constants } from '../../src/js/index.js';
+
+const { MediaUIEvents } = constants;
+
+describe('<media-controller>', () => {
+
+  it('associates itself to observe for state receivers', async () => {
+    const mediaController = await fixture(`
+      <media-controller></media-controller>
+    `);
+    assert.equal(mediaController.associatedElementSubscriptions.size, 1);
+    assert(mediaController.associatedElementSubscriptions.has(mediaController));
+  });
+
+  it('associates non-child elements to observe for state receivers', async () => {
+    const mediaController = await fixture(`
+      <media-controller id="ctrl"></media-controller>
+    `);
+    const playButton = await fixture(`
+      <media-play-button media-controller="ctrl"></media-play-button>
+    `);
+    assert.equal(mediaController.associatedElementSubscriptions.size, 2);
+    assert(mediaController.associatedElementSubscriptions.has(playButton));
+  });
+
+  it('registers itself and child controls state receivers', async () => {
+    const mediaController = await fixture(`
+      <media-controller>
+        <media-play-button></media-play-button>
+      </media-controller>
+    `);
+
+    // Also includes the media-gesture-receiver by default
+    assert.equal(mediaController.mediaStateReceivers.length, 3);
+    assert(mediaController.mediaStateReceivers.indexOf(mediaController) >= 0, 'registers itself');
+    const playButton = mediaController.querySelector('media-play-button');
+    assert(mediaController.mediaStateReceivers.indexOf(playButton) >= 0, 'registers play button');
+  });
+
+  it('registers itself and non-child controls state receivers', async () => {
+    const mediaController = await fixture(`
+      <media-controller id="ctrl"></media-controller>
+    `);
+    const playButton = await fixture(`
+      <media-play-button media-controller="ctrl"></media-play-button>
+    `);
+
+    // Also includes the media-gesture-receiver by default
+    assert.equal(mediaController.mediaStateReceivers.length, 3);
+    assert(mediaController.mediaStateReceivers.indexOf(mediaController) >= 0, 'registers itself');
+    assert(mediaController.mediaStateReceivers.indexOf(playButton) >= 0, 'registers play button');
+  });
+
+  it('registers itself and child simple element state receivers', async () => {
+    const mediaController = await fixture(`
+      <media-controller>
+        <div media-chrome-attributes="media-paused media-current-time"></div>
+      </media-controller>
+    `);
+
+    // Also includes the media-gesture-receiver by default
+    assert.equal(mediaController.mediaStateReceivers.length, 3);
+    assert(mediaController.mediaStateReceivers.indexOf(mediaController) >= 0, 'registers itself');
+    const div = mediaController.querySelector('div');
+    assert(mediaController.mediaStateReceivers.indexOf(div) >= 0, 'registers div');
+  });
+});
+
+describe('receiving state / dispatching (bubbling) events', () => {
```

Comment:
suggestion(non-blocking): You can nest `describe()`s. Since these are "more specific `<media-controller>` tests. It might be useful to have them nested in the generic `media-controller` describe, above.

Reply:
yes, I intentionally didn't nest here. I find it easier to read

Reply:
Easier to read the code or easier to read the results/output? I can see it being easier for the former, but probably not the latter? I'll add it to our eng sync discussion this week so we can just settle on one and stick to it.